module/module.go

```diff
@@ -148,14 +149,41 @@ func NewModule(ctx context.Context, address string, logger *zap.SugaredLogger) (
 	return m, nil
 }
 
+var errMissingSocketPath = errors.New("need socket path as command line argument")
+
 // NewModuleFromArgs directly parses the command line argument to get its address.
 func NewModuleFromArgs(ctx context.Context, logger *zap.SugaredLogger) (*Module, error) {
 	if len(os.Args) < 2 {
-		return nil, errors.New("need socket path as command line argument")
+		return nil, errMissingSocketPath
 	}
 	return NewModule(ctx, os.Args[1], logger)
 }
 
+// Arguments for the module executable.
+type Arguments struct {
+	Debug bool     `flag:"debug"`
+	Extra []string `flag:",extra"`
+}
+
+// NewLoggerFromArgs can be used to create a golog.Logger at either
+// "DebugLevel" or "InfoLevel" if -debug is or is not provided in os.Args
+// respectively. It will ignore all other provided arguments.
+func NewLoggerFromArgs(moduleName string) (golog.Logger, error) {
+	if len(os.Args) < 2 {
+		return nil, errMissingSocketPath
+	}
+
+	var argsParsed Arguments
+	if err := utils.ParseFlags(os.Args[1:], &argsParsed); err != nil {
+		return nil, err
+	}
+
+	if argsParsed.Debug {
+		return golog.NewDebugLogger(moduleName), nil
+	}
+	return golog.NewDevelopmentLogger(moduleName), nil
+}
```

Comment:
```suggestion
// NewLoggerFromArgs can be used to create a golog.Logger at either
// "DebugLevel" or "InfoLevel" if -debug is or is not provided in os.Args
// respectively. It will ignore all other provided arguments.
func NewLoggerFromArgs(moduleName string) golog.Logger {
	if len(os.Args) >= 3 && strings.HasSuffix(os.Args[2], "-debug") {
		return golog.NewDebugLogger(moduleName)
	}
	return golog.NewDevelopmentLogger(moduleName)
}
```

This catches both -debug and --debug options, and doesn't need to return an error, so can be inlined to the contextual main just like the previously used golog constructor was.

Reply:
I think this works for now, as the socket has to be the first arg (for backward compatibility with previous modules.) If more options are ever added though, a quick loop to just match any arg should be enough.

Reply:
Yeah this is much better for now; thanks.